File: 1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.java
class Solution {
    public int numOfSubarrays(int[] arr, int k, int threshold) {
        int count=0;
        int currentSum=0;
        int avg=0;

        for(int i=0;i<k;i++)
        {
          currentSum+=arr[i];
        }
        avg=currentSum/k;
        if(avg>=threshold){
            count++;
        }

        for(int i=k;i<arr.length;i++)
        {
            int avgg=0;
            currentSum+=arr[i]-arr[i-k];
            avgg=currentSum/k;
            if(avgg>=threshold)
            {
                count++;
            }
        }

        return count;
        
    }
}

File: 1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.py
class Solution:
    def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:
        numOfSubArray=0
        sum=0
        for i in range(0,k):
            sum=sum+arr[i]
        if sum/k>=threshold:
            numOfSubArray+=1
        for i in range(k,len(arr)):
            sum=sum+arr[i]
            sum=sum-arr[i-k]
            if sum/k>=threshold:
                 numOfSubArray+=1
        return numOfSubArray



        